pkg/oci/build.go

```diff
@@ -193,6 +192,10 @@ func createManifestForTarget(ctx context.Context, target oras.Target, metadataFi
 		return ocispec.Descriptor{}, fmt.Errorf("creating and pushing eBPF descriptor: %w", err)
 	}
 
+	// artifactType must be only set when the config.mediaType is set to
+	// MediaTypeEmptyJSON. In our case, when the metadata file is not provided:
+	// https://github.com/opencontainers/image-spec/blob/f5f87016de46439ccf91b5381cf76faaae2bc28f/manifest.md?plain=1#L170
+	var artifactType string
```

Comment:
I'm 6 months late: Is it really true? The third example in https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage has both defined, config.mediaType and artifactType. Can't we set artifactType in all cases?

Reply:
> artifactType must be **only** set when the config.mediaType is set to MediaTypeEmptyJSON

I think the "only" in the sentence above implies that artifactType can only be set when config.mediaType is empty.

> BTW, why did you come back to this discussion? Did you find any issue?

I found while reading more to implement #3372 and I was curious about it as my understanding was that we can set it on all cases (and we agree on this one)

Reply:
> > artifactType must be **only** set when the config.mediaType is set to MediaTypeEmptyJSON
> 
> I think the "only" in the sentence above implies that artifactType can only be set when config.mediaType is empty.
> 
Nope. The "only" in the sentence above implies that artifactType ~can~ must only be set when config.mediaType is empty, which gives you the possibility to do it whenever you want, but you must do it only in that case... Anyway, we understood each other 😅

> > BTW, why did you come back to this discussion? Did you find any issue?
> 
> I found while reading more to implement #3372 and I was curious about it as my understanding was that we can set it on all cases (and we agree on this one)
>
OK 🙂

Reply:
Opened #3469